src/main/kotlin/io/quarkus/code/QuarkusProjectCreator.kt

```diff
@@ -30,40 +38,52 @@ open class QuarkusProjectCreator {
                 val sourceType = CreateProject.determineSourceType(project.extensions)
                 val context = mutableMapOf("path" to (project.path as Any))
                 val success = CreateProject(zipWriter)
-                    .groupId(project.groupId)
-                    .artifactId(project.artifactId)
-                    .version(project.version)
-                    .sourceType(sourceType)
-                    .buildTool(BuildTool.MAVEN)
-                    .className(project.className)
-                    .doCreateProject(context)
+                        .groupId(project.groupId)
+                        .artifactId(project.artifactId)
+                        .version(project.version)
+                        .sourceType(sourceType)
+                        .buildTool(project.buildTool)
+                        .className(project.className)
+                        .doCreateProject(context)
                 if (!success) {
                     throw IOException("Error during Quarkus project creation")
                 }
-                AddExtensions(zipWriter, BuildTool.MAVEN)
-                    .addExtensions(project.extensions)
-                this.addMvnw(zipWriter)
+                AddExtensions(zipWriter, project.buildTool)
+                        .addExtensions(project.extensions)
+                if (project.buildTool == BuildTool.MAVEN) {
+                    addMvnw(zipWriter)
+                } else if (project.buildTool == BuildTool.GRADLE) {
+                    addGradlew(zipWriter)
+                }
             }
         }
         return baos.toByteArray()
     }
 
     private fun addMvnw(zipWrite: CommonsZipProjectWriter) {
-        zipWrite.mkdirs(MAVEN_WRAPPER_DIR)
-        writeResourceFile(zipWrite, MAVEN_WRAPPER_JAR)
-        writeResourceFile(zipWrite, MAVEN_WRAPPER_PROPS)
-        writeResourceFile(zipWrite, MAVEN_WRAPPER_DOWNLOADER)
-        writeResourceFile(zipWrite, MVNW_CMD, true)
-        writeResourceFile(zipWrite, MVNW, true)
+        zipWrite.mkdirs(MVNW_WRAPPER_DIR)
+        writeResourceFile(zipWrite, MVNW_RESOURCES_DIR, MVNW_WRAPPER_JAR)
+        writeResourceFile(zipWrite, MVNW_RESOURCES_DIR, MVNW_WRAPPER_PROPS)
+        writeResourceFile(zipWrite, MVNW_RESOURCES_DIR, MVNW_WRAPPER_DOWNLOADER)
+        writeResourceFile(zipWrite, MVNW_RESOURCES_DIR, MVNW_CMD, true)
+        writeResourceFile(zipWrite, MVNW_RESOURCES_DIR, MVNW, true)
+    }
+
+    private fun addGradlew(zipWrite: CommonsZipProjectWriter) {
```

Comment:
wait - code.quarkus.io does its own writing and creation of the mvn/gradle wrappers ? 
we should put this stuff into the base tooling rather than code.quarkus.io duplicates it.

for this specific issue of getting gradlew enabled on code.quarkus.io + 1 though. just something we should get cleaned up.

Reply:
I remember @ia3andy pointing an issue where the Maven wrapper wasn't created in the devtools-common, but I can't find the issue, maybe that's already fixed

Reply:
Yes, and it is still not created there - and the mvn mojo does call outs to mvn/gradle with "random" success - I found that the other day and started to wonder what kind of funky differences we have :)

But yeah, I think we should move this code into devtools-common and tell users to run `gradle wrapper --upgrade` and if similar exist for `mvn` wrapper.

Reply:
Agreed, feel free to copy those files over to devtools-common, since you're already working on it :wink:

Reply:
Steps to upgrade the `mvn`wrapper: https://github.com/takari/takari-maven-plugin#updating-the-maven-wrapper-in-your-maven-project

TLDR: `mvn -N io.takari:maven:0.7.6:wrapper` and it will overwrite the installed files

Reply:
Found the issue: https://github.com/quarkusio/code.quarkus.io/issues/15